gcs_oauth2_boto_plugin/oauth2_helper.py

```diff
@@ -100,6 +100,10 @@ def OAuth2ClientFromBotoConfig(config,
     client_secret = config.get('OAuth2', 'client_secret',
                                os.environ.get('OAUTH2_CLIENT_SECRET',
                                               CLIENT_SECRET))
+    ca_certs_file=config.get_value('Boto', 'ca_certificates_file')
+    if ca_certs_file == 'system':
+      ca_certs_file = None
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 01/31/2015 07:34 PM, Mike Schwartz wrote:
>> @@ -100,6 +100,10 @@ def OAuth2ClientFromBotoConfig(config, 
>> client_secret = config.get('OAuth2', 'client_secret', 
>> os.environ.get('OAUTH2_CLIENT_SECRET', CLIENT_SECRET)) +
>> ca_certs_file=config.get_value('Boto', 'ca_certificates_file') +
>> if ca_certs_file == 'system': +      ca_certs_file = None
> 
> Can you describe the use case where we'd want to fall through to
> the system certificates file?

All the time. Linux distributions generally have a security team that
watches out for certificate related issues as well as security issues
in distributed code. I suspect a similar mechanism for OS X, but have
no idea how certificates are in Windoze.

Members of these teams are on the embargo list and thus can react and
prepare new packages ahead of the disclosure. Then when the disclosure
hits new packages are available. A sysadmin simply updates the
packages and everything is in good working order.

In a case where a system admin needs to configure and manage
certificates that are provided by applications not only is the systems
administrator saddled with a certificate management nightmare, but in
case of  a certificate breach the systems administrator would also
have to wait for the application to provide new certificates or figure
out how to fiddle with the application configuration to make it happy
to work with another set of certificates (this could be simply
pointing the config to another directory, or the admin might have to
create a file, or ....) For the most part application developers are
not on the embargo list and thus the reaction time is slower.
Additionally a sysadmin would have to know all the places where
certificates are configured for each application. I could easily
continue this nightmare from 5 different angle.

The second scenario is, at least from my point of view, inferior to
the first approach.

Anyway, boto supports the 'system" keyword and thus this code should
properly handle the 'system' keyword rather than passing 'system' to
the underlying http implementation as certificate location.



- -- 
Robert Schweikert                           MAY THE SOURCE BE WITH YOU
Public Cloud Architect                                 LINUX
rjschwei@suse.com
IRC: robjo
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2.0.22 (GNU/Linux)

iQEcBAEBAgAGBQJUzhIfAAoJEE4FgL32d2Ukf/IH/1POBXrxpKEc3+dq9nhjyaJO
Hvf7940rHGi40TeOvFU5QZSC3hORD0mamleEelBT5oIGdVRMTnC2Q2SkehK8uDT7
xVjAbLe11tv8NAu92UdkZmCee8SV4FIyzIfeIJPFPeoZYcjht785r3A/5iUv3FGh
Saieu8Urr/U6CWtdOYbyczGRkhMCxq4tOwGtt24lvVm0EwzR3qbdfB/w5EpiIF+i
bPQ/w+XC9jeBlTST/ji0PwJmDIVPNBleZawMBMaQkWcTe6FVZLn5zPccvaC0dmZC
KVSRNarYw3NjBrkaJGE99WqrweiGbifzMMFIGz+kUbpYnK5qV9G6D27XvqrPCvM=
=ElDY
-----END PGP SIGNATURE-----